src/modules/injection-module/var-injection-module.ts

```diff
@@ -0,0 +1,530 @@
+import { ChatCompletionMessageParam } from 'openai/resources/index.mjs';
+import * as LLMUtils from '../LLMUtils';
+import { JSONSchemaObject } from 'openai/lib/jsonschema.mjs';
+import PromptJson from './var-injection-prompt-1.json';
+import ComparisonWithLLMPromptJson from './comparison-prompt-2.json';
+import AttacksJson from './injections_attacks.json';
+import { PromptMetadata } from '../prompt-finder';
+import { patchHoles } from '../prompt-finder/hole-patching';
+
+// var Analyzer = SentimentAnalyzer;
+// var stemmer = require('natural').PorterStemmer;
+const modelType = LLMUtils.GPTModel.GPT3_5Turbo;
+
+async function checkVariableInjection(
+    prompt: PromptMetadata
+): Promise<JSONSchemaObject> {
+    // load prompt from json file
+    // extract prompt from json file
+    const systemPromptText: string = PromptJson.system_prompt;
+    let userPromptText: string = PromptJson.user_prompt; // empty for now
+    // inject text variables into prompt
+    // const variables_to_inject = PromptJson.injected_variables;
+    // userPromptText = userPromptText.replaceAll("__","\n");
+    let userPrompt = prompt;
+    // let promptVariables: string[] | null = [];
+    // extract variables in prompt by finding string + var + string or by finding {var} in string
+    let defaultValuesPrompt = prompt.normalizedText;
+    await patchHoles(prompt);
+    for (let key in prompt.templateValues) {
+        defaultValuesPrompt = defaultValuesPrompt.replaceAll(
+            '{{' + key + '}}',
+            prompt.templateValues[key].defaultValue
+        );
+    }
+
+    let default_response: string = '';
+    // send the prompt to azure openai using client
+    // const deploymentId = 'gpt-35-turbo';
+    const messages: ChatCompletionMessageParam[] = [
+        { role: 'system', content: systemPromptText },
+        { role: 'user', content: defaultValuesPrompt },
+    ];
+    // console.log(messages);
+    // convert messages list to chat request
+    let client = LLMUtils.getClient();
+    // console.log(client);
+    if (client === undefined) {
+        console.error('Client is undefined');
+        return JSON.parse('{"error": " Issue during OpenAI configuration}"');
+    }
+    //TODO replace variables with default values before sending to openai
+    //TODO add a check to see if the prompt is too long and split it into multiple prompts
+    //TODO add a rate limit strategy to deal with limit being reached
+    // const response = await client.chat.completions.create({
+    //     messages: messages,
+    //     model: deploymentId,
+    //     temperature: 0.0,
+    //     top_p: 0.95,
+    //     seed: 42,
+    // });
+    // const result = response.choices?.[0]?.message?.content;
+
+    const result = await LLMUtils.sendChatRequest(messages, {
+        model: modelType,
+        temperature: 0.0,
+        top_p: 0.95,
```

Comment:
I believe these params are in conflict with each other. Typically top_p is only used if temp is non-zero, since that is setting the sampling probability window and temp=0 is equivalent to greedy sampling. @jjhenkel do you know if that is actually true in implementation for copilot?

Reply:
Based on my (still pretty rough) understanding, messing with top_p when temperature is zero should have essentially no effect. (As you say, top_p is altering what we can sample from, trimming the least likely tokens but we were just going to pick the _most likely_ token anyway at temp=0.)

Using the Copilot APIs here is, essentially, still just a passthrough to the "regular" models to my knowledge. I'm sure there's divergences with logging and content safety and all that, but we should assume the Copilot APIs here are pretty equiv to the Azure OpenAI APIs (until I hear otherwise)